partcad/src/partcad/part_factory_scad.py

```
@@ -86,3 +89,108 @@ async def instantiate(self, part):
+        base_dir = os.path.dirname(self.path)
+        if not base_dir:
+            base_dir = "."
+
+        lock_path = "/tmp/partcad.lock"
+        with open(lock_path, "w") as lock_file:
+            fcntl.lockf(lock_file, fcntl.LOCK_EX)
+
+            try:
+                if ".partcad/git" in base_dir.replace("\\", "/"):
+                    # Create a stable name for the SCAD file
+                    hashed = hashlib.sha1(base_dir.encode("utf-8")).hexdigest()
```

Ruff (0.8.2)

Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)